### DISCUSSION TITLE
Fault Activity => how to catch errors and propagate them out

### DISCUSSION THREAD

**Discussion by fulganta** [elsa-workflows/elsa-core#2885] [Design & tech]
Hi,

I have a workflow that at the end reply to the client via http.
Issue is in case an activity fail, well no answer is sent. So I want that if any activity of my workflow fail, the end client is aware abut it.
Do you have such a logic implemented?
Otherwise, how can I catch the "fault" value out of my custom activity? Currently I just catch the "Done" value.
Thanks in advance for your help

**Comment by sfmskywalker** [comment]
Makes sense.

Elsa 2 doesn't have a "catch any" activity, but Elsa3 will have this.

For Elsa 2, the best thing you can do is implement a custom handler in your application that handles the `WorflowFaulted` event.

For example:

```csharp
public class MyWorkflowFaultHandler : INotificationHandler<WorkflowFaulted>
{
   public Task Handle(WorkflowFaulted notification, CancellationToken cancellationToken)
   {
      // Your HTTP call here.
   }
}
```

You can register your handlers like this:

```
services.AddNotificationHandlersFrom<MyWorkflowFaultHandler>(); // Registers all handlers from the assembly containing the specified type.
```

**Reply by ekram0** [reply]
@sfmskywalker 

it's a good, but i have an issue.
incase implement parent child workflow and issue happened in child. 
this method run 4 time which's replicate exe logs.

how to prevent this from running n times of execution.

elsa version: 2.x

thanks
